### DISCUSSION TITLE
Create solution from AuxVariables

### DISCUSSION THREAD

**Discussion by matthiasneuner** [idaholab/moose#21746] [Q&A General] [answered]
Assuming that 2 sub apps (all identical meshes, also the main app) are computing 2 solutions,
I would like to add (or subtract) them in an AuxVariable, and set them as a new solution for the next iteration step, e.g.,

```
  / - > solution_n -> sub_app_1 -> sub_sol_1 
 /  - > solution_n -> sub_app_2 -> sub_sol_2
|
|                       V
|
|        aux_sol = sub_sol_1 + sub_sol_2
|
|                       V  
|
 \---- solution_n+1 = aux_sol
```

Is that possible in Moose? Which modules would be relevant for this case?
In particular, how can I set the current solution from the AuxVariable?

Thank you in advance!

**Comment by GiudGiud** [comment, marked as answer]
Hello

You could:
1)
- write your own transfer which does that directly
2)
OR (I am assuming by solution you mean a NON LINEAR, eg a main, variable here btw)
- add an auxvariables for storage on sub app 1
- transfer the other auxvariable from subapp 2 to subapp 1, using the CopyTransfer since you have the same meshes
- use a [ParsedAux](https://mooseframework.inl.gov/source/auxkernels/ParsedAux.html) to compute the difference and store it in a third variable on subapp1
- transfer it to the main app to overwrite the initial condition
The tricky thing is that the subapps will have to be executed on different schedules for this to work, because we currently do not support this: `execute main app, execute app2, transfer from 2 to 1, execute app1`. So use TIMESTEP_BEGIN/END/ other execute_on to stagger everything in the right order. 
Please consider using the `verbose_multiapps` parameter of the FEProblem to check the execution order. 
Here we are working around the fact that auxkernels cannot initialize main variables 

3)
OR
- add two auxvariables for storage on main app
- transfer the auxvariables from subapps to main app, using the CopyTransfer since you have the same meshes
- write a user object yourself (should not be too hard in terms of UOs) that does the difference and stores in the nonlinear variable
This has the advantage that the execution schedule is a lot more natural. But you ll have to write code

Hope this helps,
Guillaume

**Reply by matthiasneuner** [reply]
Thank you very much; it seems approach 3) is the way to go.

I think that using such an approach my main_app actually needs no 'nonlinear variables' at all, since the main app should not solve directly.
Currently, this gives me an error message ` No variables specified in the FEProblemBase 'MOOSE Problem'`.
I hope that this is supported, but I will create another question for this.

**Reply by GiudGiud** [reply]
oh the main app isnt solving a NL system?

Then just use a ParsedAux in option 3) to store the difference in the 3rd AuxVariable

**Reply by matthiasneuner** [reply]
Yes, I think that will be the best solution :-) . Thank you again!
